internal/runtime/client/client.go

```diff
@@ -162,8 +161,20 @@ func (h *hookClient) Call(ctx context.Context, name string, request, response ru
 		timeout:       timeoutDuration,
 		failurePolicy: registration.FailurePolicy,
 	}
-	if err := httpCall(ctx, request, response, opts); err != nil {
-		return errors.Wrapf(err, "failed to call extension '%s'", name)
+	err = httpCall(ctx, request, response, opts)
+	if err != nil {
+		return errors.Wrap(err, "failed to call extension")
+	}
+	failure, err := runtimehooksv1.IsFailure(response)
+	if err != nil {
+		return errors.Wrap(err, "failed to process response")
+	}
+	if failure {
+		msg, err := runtimehooksv1.GetMessage(response)
+		if err != nil {
+			return errors.Wrap(err, "failed to process response")
+		}
+		return &runtimehooksv1.ErrExtensionFailure{ExtensionName: name, Message: msg}
```

Comment:
Q: The plan here was to return the error in a way that folks can figure out which extension exactly failed, right? (although we only use it in Call so it should be already clear which one it is)

I wonder if either:
a) we don't need the custom error for now (we can always introduce it later and return a regular error for now)
b) the error should be defined in the client package as we only use it here (it seems to be an error of the Client not something that we have to expose as part of the hook API)